File: api/subagents.js
// Vercel API Route for subagent registry with garage-mcp integration
export default async function handler(req, res) {
    if (req.method !== 'GET') {
        return res.status(405).json({ error: 'Method not allowed' })
    }
    
    const BASE = process.env.GARAGE_MCP_URL
    const TOKEN = process.env.GARAGE_MCP_TOKEN
    const PATH = process.env.SUBAGENT_REGISTRY_PATH || '/registry/subagents'
    
    const FALLBACK = [
        {"id":"validate-ssot","bay":"frontend","desc":"Validate SSOT against HEIR schema"},
        {"id":"heir-check","bay":"backend","desc":"Run HEIR checks on blueprint"},
        {"id":"register-blueprint","bay":"backend","desc":"Persist + emit registration event"},
    ]
    
    // If no garage-mcp URL configured, return fallback
    if (!BASE) {
        return res.status(200).json({ items: FALLBACK })
    }
    
    try {
        const headers = {
            'Content-Type': 'application/json',
            ...(TOKEN && { 'Authorization': `Bearer ${TOKEN}` })
        }
        
        const response = await fetch(`${BASE}${PATH}`, {
            method: 'GET',
            headers,
            timeout: 5000
        })
        
        if (!response.ok) {
            throw new Error(`HTTP ${response.status}: ${response.statusText}`)
        }
        
        const data = await response.json()
        const items = Array.isArray(data) ? data : (data.items || [])
        
        const processedItems = items.map(item => ({
            id: item.id || item.name || 'unknown',
            bay: item.bay || item.namespace || 'unknown',
            desc: item.description || item.desc || ''
        }))
        
        // Return processed items or fallback if empty
        return res.status(200).json({ 
            items: processedItems.length > 0 ? processedItems : FALLBACK 
        })
        
    } catch (error) {
        console.error('Garage-MCP fetch error:', error.message)
        // Gracefully fall back to static list
        return res.status(200).json({ items: FALLBACK })
    }
}

File: src/server/infra/subagents.py
import os, httpx

BASE = os.getenv("GARAGE_MCP_URL", "")
TOKEN = os.getenv("GARAGE_MCP_TOKEN", "")
PATH = os.getenv("SUBAGENT_REGISTRY_PATH", "/registry/subagents")

FALLBACK = [
    {"id":"validate-ssot","bay":"frontend","desc":"Validate SSOT against HEIR schema"},
    {"id":"heir-check","bay":"backend","desc":"Run HEIR checks on blueprint"},
    {"id":"register-blueprint","bay":"backend","desc":"Persist + emit registration event"},
]

def list_subagents():
    if not BASE:
        return FALLBACK
    try:
        headers = {"Authorization": f"Bearer {TOKEN}"} if TOKEN else {}
        with httpx.Client(base_url=BASE, headers=headers, timeout=5.0) as c:
            r = c.get(PATH)
            r.raise_for_status()
            data = r.json()
            items = data if isinstance(data, list) else data.get("items", [])
            out = []
            for item in items:
                out.append({
                    "id": item.get("id") or item.get("name"),
                    "bay": item.get("bay") or item.get("namespace") or "unknown",
                    "desc": item.get("description") or "",
                })
            return out or FALLBACK
    except Exception:
        return FALLBACK